frontend/catalyst/api_extensions/control_flow.py

```diff
@@ -549,7 +549,8 @@ def main():
     """
 
     def __init__(self, pred, true_fn):
-        self.preds = [pred]
+        # Make sure the predicate is a boolean
+        self.preds = [jnp.astype(pred, bool)]
```

Comment:
We also need to do the same verification for other predicates added in elif statements

Reply:
Done